diff --git a/dist/runtime/server/render/component.js b/dist/runtime/server/render/component.js
index 26d6c8b..fd28f57 100644
--- a/dist/runtime/server/render/component.js
+++ b/dist/runtime/server/render/component.js
@@ -1,6 +1,6 @@
 import { createRenderInstruction } from "./instruction.js";
 import { AstroError, AstroErrorData } from "../../../core/errors/index.js";
-import { markHTMLString } from "../escape.js";
+import { escapeHTML, markHTMLString } from "../escape.js";
 import { extractDirectives, generateHydrateScript } from "../hydration.js";
 import { serializeProps } from "../serialize.js";
 import { shorthash } from "../shorthash.js";
@@ -276,7 +276,7 @@ async function renderFrameworkComponent(result, displayName, Component, _props, 
     unrenderedSlots = Object.keys(children);
   }
   const template = unrenderedSlots.length > 0 ? unrenderedSlots.map(
-    (key) => `<template data-astro-template${key !== "default" ? `="${key}"` : ""}>${children[key]}</template>`
+    (key) => `<template data-astro-template${key !== "default" ? `="${escapeHTML(key)}"` : ""}>${children[key]}</template>`
   ).join("") : "";
   island.children = `${html ?? ""}${template}`;
   if (island.children) {
